elf/compat.go

```diff
@@ -22,33 +22,25 @@ import (
 	"runtime"
 )
 
-func GetSyscallFnName(name string) (string, error) {
+func GetSyscallFnName(name string, symFile string) (string, error) {
```

Comment:
I'd leave the old signature for the public interface and use a private function for testing, that way if someone wants to use this function, they don't need to remember that the sym file is /proc/kallsyms, something like this:

```
const defaultSymFile = "/proc/kallsyms"
func GetSyscallFnName(name string) (string, error) {
  return getSyscallFnNameWithFile(name, defaultSymFile)
}

func getSyscallFnNameWithFile(name string, symFile string) (string, error) {
  // the full implementation here
```